### PR TITLE
Add Storage for Attendance

### PR DESCRIPTION
**Pull request by jeunhoe**
Hi team, implemented the storage for Attendance. Started on this awhile back so I think there will be quite a large number of merge conflicts. Do help me take a look at it!

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -90,7 +97,21 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
             initialData = new AddressBook();
         }
 
-        return new ModelManager(initialData, userPrefs);
+        try {
+            attendanceOptional = storage.readAttendance();
+            if (!attendanceOptional.isPresent()) {
+                logger.info("Data file not found. Will be starting with a sample AddressBook");
```
Comment (hellodommy) [inline src/main/java/seedu/address/MainApp.java:103]: Is this the case when the user has not recorded any attendance?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -30,20 +30,20 @@
     /**
      * Initializes a ModelManager with the given addressBook and userPrefs.
      */
-    public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs) {
+    public ModelManager(ReadOnlyAddressBook addressBook, Attendance attendance, ReadOnlyUserPrefs userPrefs) {
```
Comment (hellodommy) [inline src/main/java/seedu/address/model/ModelManager.java:33]: Changing this constructor will require updating `FilterCommandTest` and `SortCommandTest`as well.

**Review by hellodommy — COMMENTED** [review]
Good job Jeunhoe on implementing the storage for training/attendance.

We will have to update some classes the fix the failing tests.

File: src/main/java/seedu/address/MainApp.java (hunk 1 above)
Comment (jeunhoe) [reply src/main/java/seedu/address/MainApp.java:103]: Yup! If it was unable to obtain the attendance.json file, a new one will be automatically created.

File: src/main/java/seedu/address/model/ModelManager.java (hunk 2 above)
Comment (jeunhoe) [reply src/main/java/seedu/address/model/ModelManager.java:33]: Done! I've pulled the from the remote branch and updated the failing tests. Turns out TrainingTest and SelectCommandTest were broken too!

File: src/main/java/seedu/address/logic/commands/TrainingCommand.java [hunk 3]
```diff
@@ -28,15 +30,15 @@
...
     private List<Index> indexList;
 
     public TrainingCommand(List<Index> indexList) {
-        this.date = new Date();
+        this.date = new Date().toString();
         this.indexList = indexList;
     }
 
-    public TrainingCommand(Date date, List<Index> indexList) {
+    public TrainingCommand(String date, List<Index> indexList) {
```
Comment (hellodommy) [inline src/main/java/seedu/address/logic/commands/TrainingCommand.java:41]: Just curious - why are there 2 constructors for `TrainingCommand`?

When is which constructor used?

**Pull request merged**